### PR TITLE
320 be spike check how to implement a user of type recruiter

### PR DESCRIPTION
**Pull request by ivanlegranbizarro**

### CODE REVIEW

File: app/Models/Recruiter.php [hunk 1]
```diff
@@ -0,0 +1,11 @@
+<?php
+
```
Comment (jordimorillo) [inline app/Models/Recruiter.php:2]: declare(strict_types=1);

File: database/migrations/2024_10_19_172804_create_companies_table.php [hunk 2]
```diff
@@ -0,0 +1,31 @@
+<?php
+
```
Comment (jordimorillo) [inline database/migrations/2024_10_19_172804_create_companies_table.php:2]: declare(strict_types=1);

File: database/migrations/2024_10_19_173835_create_recruiters_table.php [hunk 3]
```diff
@@ -0,0 +1,31 @@
+<?php
+
```
Comment (jordimorillo) [inline database/migrations/2024_10_19_173835_create_recruiters_table.php:2]: declare(strict_types=1);

File: database/migrations/2024_10_19_174009_create_job_offers_table.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
+<?php
+
```
Comment (jordimorillo) [inline database/migrations/2024_10_19_174009_create_job_offers_table.php:2]: declare(strict_types=1);

File: database/migrations/2024_10_19_172804_create_companies_table.php [hunk 5]
```diff
@@ -11,10 +11,10 @@
     public function up(): void
     {
         Schema::create('companies', function (Blueprint $table) {
-            $table->id();
+            $table->uuid('id')->primary();
             $table->string('name', 255);
-            $table->string('email', 255);
-            $table->string('NIF', 255);
+            $table->string('email', 255)->unique();
```
Comment (jordimorillo) [inline database/migrations/2024_10_19_172804_create_companies_table.php:16]: what happens if a person has an email and uses the same email in all the companies he has?

File: database/migrations/2024_10_20_072243_create_job_offer_student_table.php [hunk 6]
```diff
@@ -0,0 +1,29 @@
+<?php
+
```
Comment (jordimorillo) [inline database/migrations/2024_10_20_072243_create_job_offer_student_table.php:2]: declare(strict_types=1);

File: database/migrations/2024_10_20_072243_create_job_offer_student_table.php [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+use App\Models\JobOffer;
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+return new class extends Migration {
+    /**
```
Comment (jordimorillo) [inline database/migrations/2024_10_20_072243_create_job_offer_student_table.php:9]: remove the comments

**Review by StephaneCarteaux — APPROVED** [review]

**Pull request closed without merging**